### ISSUE TITLE
Add ability to flatten included dependencies

### ISSUE THREAD

**Issue by kvverti** [FabricMC/fabric-loom#215]
Currently, included JIJ dependencies may have nested JIJ. This can result in multiple JIJ dependencies having the same nested JIJ dependency. Loom should include an option to flatten and de-duplicate JIJ dependencies, similar to what Fabric Loader does at mod load time.

This issue is part of [a plan to improve depending on Fabric API](https://gist.github.com/kvverti/3fc6a05d974210f785a9a751d5e0ae28).

**Comment by Pyrofab** [comment]
This would definitely be useful for any library, not just Fabric API.

**Comment by modmuss50** [comment]
I dont think this is currenly an issue with fabric api specifically, it can be manually worked around by being careful with what you include.

It would take a bit of work as it would need a whole chunk of fabric-loaders mod resoultion stuff pulled over.
